raiden-ts/src/raiden.ts

```diff
@@ -259,6 +263,26 @@ export class Raiden {
       return { totalSupply, decimals, name, symbol };
     });
 
+    this.getUDCBalance = async () =>
+      from(this.deps.userDepositContract.functions.balances(address))
+        .pipe(
+          map(balance => {
+            const owedAmount = Object.values(state.path.iou)
+              .reduce(
+                (acc, value) =>
+                  acc.concat(
+                    Object.values(value).filter(value =>
+                      value.expiration_block.gte(state.blockNumber),
+                    ),
+                  ),
+                new Array<IOU>(),
+              )
+              .reduce((acc, iou) => acc.add(iou.amount), Zero);
```

Comment:
Might be better to map-reduce instead of reducing twice

Reply:
Could you give me a basic example of how it would work? I tried a few approaches and I don't think I get it. 

One reduce is used to flatten an array of arrays to a single dimension array (instead of flatMap), and the other reduce is used to calculate the sum.

Reply:
Hmm I think map isn't even necessary, my mistake. It could simply be
```
Object.values(state.path.iou)
   .flat()
   .filter(iou => iou.expiration_block.gte(state.blockNumber))
   .reduce((acc,iou) => ...)
```
(untested)

Reply:
Do you know if `flat` is poly filled when it is not supported I can see that it is implemented around 83% of the browsers.

I was trying flatMap didn't even think of flat (and flatMap didnt work)

Reply:
Hmm I'm not sure if it's polyfilled, but looks like it's supported by all browsers that MetaMasks supports. I guess that's enough support for us? Should work just fine on nodejs too. Maybe we can bring this up internally to get clarification

Reply:
I will give it a try. It shouldn't take much time :)